packages/atomic/src/components/commerce/atomic-commerce-facet/atomic-commerce-facet.spec.ts

```diff
@@ -0,0 +1,623 @@
+    },
+    getFacetValueButtonByLabel(value: string | RegExp) {
+      return page.getByLabelText(`Inclusion filter on ${value}`);
+    },
+    get componentError() {
+      return page.getByText(
+        'Look at the developer console for more information'
+      );
+    },
+    parts: (element: Element) => {
+      const qs = (part: string) =>
+        element.shadowRoot?.querySelector(`[part~="${part}"]`);
+      const qsa = (part: string) =>
+        element.shadowRoot?.querySelectorAll(`[part~="${part}"]`);
+      return {
+        facet: qs('facet'),
+        labelButton: qs('label-button'),
+        labelButtonIcon: qs('label-button-icon'),
+        clearButton: qs('clear-button'),
+        clearButtonIcon: qs('clear-button-icon'),
+        searchWrapper: qs('search-wrapper'),
+        searchInput: qs('search-input'),
+        searchIcon: qs('search-icon'),
+        searchClearButton: qs('search-clear-button'),
+        moreMatches: qs('more-matches'),
+        noMatches: qs('no-matches'),
+        matchesQuery: qs('matches-query'),
+        searchHighlight: qs('search-highlight'),
+        values: qsa('values'),
+        valueLabel: qsa('value-label'),
+        valueCount: qsa('value-count'),
+        valueCheckbox: qsa('value-checkbox'),
+        valueCheckboxChecked: qsa('value-checkbox-checked'),
+        valueCheckboxLabel: qsa('value-checkbox-label'),
+        valueCheckboxIcon: qsa('value-checkbox-icon'),
+        showMore: qs('show-more'),
+        showLess: qs('show-less'),
+        showMoreLessIcon: qs('show-more-less-icon'),
+      };
+    },
+  };
+
+  beforeEach(() => {
+    mockedConsoleError = vi
+      .spyOn(console, 'error')
+      .mockImplementation(() => {});
+
+    mockedSummary = buildFakeSummary({});
+    mockedFacet = buildFakeRegularFacet({
+      implementation: {
+        toggleSelect: vi.fn(),
+        toggleExclude: vi.fn(),
+        deselectAll: vi.fn(),
+        showMoreValues: vi.fn(),
+        showLessValues: vi.fn(),
+        facetSearch: {
+          clear: vi.fn(),
+          updateText: vi.fn(),
+          search: vi.fn(),
+          select: vi.fn(),
+          exclude: vi.fn(),
+          singleExclude: vi.fn(),
+          singleSelect: vi.fn(),
+        },
+      },
+    });
+  });
+
+  afterEach(() => {
+    mockedConsoleError.mockRestore();
+  });
+
+  const setupElement = async ({isCollapsed} = {isCollapsed: false}) => {
+    const {element} =
+      await renderInAtomicCommerceInterface<AtomicCommerceFacet>({
+        template: html`<atomic-commerce-facet
+          .facet=${mockedFacet}
+          .summary=${mockedSummary}
+          ?isCollapsed=${isCollapsed}
+          field="testField"
+        ></atomic-commerce-facet>`,
+        selector: 'atomic-commerce-facet',
+        bindings: (bindings) => {
+          bindings.store.getUniqueIDFromEngine = vi.fn().mockReturnValue('123');
+          return bindings;
+        },
+      });
+    return element;
+  };
+
```

Comment:
We have pretty good test coverage for rendering, but we have nothing for lifecycle.

I think we should test the `initialize` and `disconnectedCallback`.